src/frontend/js/components/Dashboard/DashboardBreadcrumbs.tsx

```diff
@@ -0,0 +1,123 @@
+import { Button } from 'components/Button';
+import { Icon } from 'components/Icon';
+import { IntlHelper } from 'utils/IntlHelper';
+import { location } from 'utils/indirection/window';
+
+export interface DashboardBreadcrumbsMeta {
+  placeholders: DashboardBreadcrumbsPlaceholders;
+}
+
+export type DashboardBreadcrumbsPlaceholders = Record<string, string>;
+
+const messages = defineMessages({
+  back: {
+    id: 'components.DashboardBreadcrumbs.back',
+    description: "The dashboard's breadcrumb back button's label",
+    defaultMessage: 'Back',
+  },
+});
+
+interface DashboardBreadcrumbsContextInterface {
+  pushBreadcrumbsPlaceholders: (data: DashboardBreadcrumbsPlaceholders) => void;
+  meta: DashboardBreadcrumbsMeta;
+}
+
+export const DashboardBreadcrumbsContext = createContext<DashboardBreadcrumbsContextInterface>(
+  {} as any,
+);
+
+/**
+ * Provider that needs to wrap any component using the DashboardBreadcrumbs component.
+ * It provides a context used to make formatjs value available bottom up.
+ */
+export const DashboardBreadcrumbsProvider: FC = ({ children }) => {
+  const [placeholders, setPlaceholders] = useState<DashboardBreadcrumbsPlaceholders>({});
+
+  const pushBreadcrumbsPlaceholders = (data: DashboardBreadcrumbsPlaceholders) => {
+    setPlaceholders({ ...placeholders, ...data });
+  };
+
+  // JSON.stringify is needed because `placeholders` is an object this is the same as doing a
+  // shallow equality test. Otherwise, it will create an infinite re-rendering loop as the object
+  // reference changes on each use of pushBreadcrumbsPlaceholders.
+  const context = useMemo(
+    () => ({
+      pushBreadcrumbsPlaceholders,
+      meta: { placeholders },
+    }),
+    [JSON.stringify(placeholders)],
+  );
+
+  return (
+    <DashboardBreadcrumbsContext.Provider value={context}>
+      {children}
+    </DashboardBreadcrumbsContext.Provider>
+  );
+};
+
+/**
+ * Hook to be used in every component that needs to provide a formatjs value to breadcrumbs's part.
+ */
+export const useBreadcrumbsPlaceholders = (data: DashboardBreadcrumbsPlaceholders) => {
+  const context = useContext(DashboardBreadcrumbsContext);
+  useEffect(() => {
+    context.pushBreadcrumbsPlaceholders(data);
+  }, [data]);
+};
+
+/**
+ * Returns formatted breadcrumbs parts.
+ */
+const useBreadcrumbsParts = () => {
+  const context = useContext(DashboardBreadcrumbsContext);
+  const matches = useMatches();
+  const intl = useIntl();
+  return matches
+    .filter((match) => (match.handle as any)?.crumbLabel)
+    .filter((match) => {
+      const { crumbLabel } = match.handle as { crumbLabel: MessageDescriptor };
+      return IntlHelper.doValuesExists(crumbLabel, context.meta.placeholders);
+    })
+    .map((match) => {
+      const { crumbLabel } = match.handle as { crumbLabel: MessageDescriptor };
+      return {
+        pathname: match.pathname,
+        name: intl.formatMessage(crumbLabel, context.meta.placeholders),
+      };
+    });
```

Comment:
What do you think to use `reduce` instead of 2 `filters` and one `map` ?
```suggestion
return matches.reduce((crumbs, match) => {
    const label: MessageDescriptor = (match.handle as any)?.crumbLabel;

    if (label && IntlHelper.doValuesExists(label, context.meta.placeholders)) {
      crumbs.push({
        name: intl.formatMessage(label, context.meta.placeholders),
        pathname: match.pathname,
      });
    }

    return crumbs;
  }, [] as Record<'pathname' | 'name', string>[]);
```

Reply:
I really like this approach